pkg/storage/replica.go

```diff
@@ -994,6 +994,7 @@ func (r *Replica) leaseGoodToGo(ctx context.Context) (LeaseStatus, bool) {
 //  will not incur latency waiting for the command to complete.
 //  Reads, however, must wait.
 func (r *Replica) redirectOnOrAcquireLease(ctx context.Context) (LeaseStatus, *roachpb.Error) {
+	ctx = r.AnnotateCtx(ctx)
```

Comment:
I think only `getLeaseForGossip` hasn't annotated the context when calling `redirectOnOrAcquireLease`. Probably better to annotate the context in the call stack (e.g. in `maybeGossipFirstRange`).